### DISCUSSION TITLE
sum child attributes

### DISCUSSION THREAD

**Discussion by mpiko** [freeplane/freeplane#678] [General]
Hi,

I've added a couple of attributes to each child node for time and cost. That's pretty straight forward. 
Is it possible to use the same attributes on a parent node and have it automatically sum and display to totals of time and cost which have been applied to each of its children nodes? By doing so, the cost and duration could be quickly determined for the entire activity.
Is this possible?

**Comment by macmarrum** [comment]
Hi

yes, using formulas.

If the values in your attributes are numeric, summing all children's values can be achieved by adding the following formula to an attribute in the parent (just change 'your attribute name'):
```groovy
=node.children.collect { it['your attribute name'].num0 }.sum()
```

If you need more help, please attach a (zipped) mindmap – it can be just an example, if you'd like to avoid sharing your original mindmap.

**Comment by mpiko** [comment]
Brilliant!!! Thank you so much.
It worked perfectly.
